ballerina-shell/modules/shell-cli/src/main/java/io/ballerina/shell/cli/jline/JlineBallerinaParser.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.shell.cli.jline;
+
+import io.ballerina.shell.cli.jline.parser.ParserStateMachine;
+import org.jline.reader.EOFError;
+import org.jline.reader.ParsedLine;
+import org.jline.reader.Parser;
+import org.jline.reader.SyntaxError;
+
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * A parser for Ballerina Shell to detect incomplete lines.
+ *
+ * @since 2.0.0
+ */
+public class JlineBallerinaParser implements Parser {
+    private static final char SPACE = ' ';
+    private static final char NEW_LINE = '\n';
+
+
+    @Override
+    public ParsedLine parse(String line, int cursor, ParseContext context) throws SyntaxError {
+        assert line != null;
```

Comment:
Assertions aren't enabled by default, right? Is it possible that `line` will actually be null? 

https://stackoverflow.com/questions/2758224/what-does-the-java-assert-keyword-do-and-when-should-it-be-used

Reply:
No they are not,
but this should not get null, but I added so that test cases would fail if they did. I can remove if wanted